TypeProvider/Core/Message.fs

```diff
@@ -0,0 +1,53 @@
+namespace Froto.TypeProvider.Core
+
+open Froto.Serialization
+
+// Eventually, this class might be replaced by Froto.Core.Encoding.MessageBase, but so far
+// this interface looks simpler and satisfies all needs.
+
+/// Base class for types generated from proto messages.
+[<AbstractClass>]
+type Message() as this =
+ 
+    let mutable size = lazy (
+        let buffer = NullWriteBuffer()
+        this.Serialize buffer
+        buffer.Length
+    )
+ 
+    member this.SerializedLength = size.Value
```

Comment:
Good catch! Originally generated types meant to be immutable, but later I switched to mutable types and forgot to fix this one. Anyway, now it's fixed + there's a separate unit test for that case.